zetaclient/zetacore/broadcast.go

```diff
@@ -107,6 +111,16 @@ func (c *Client) Broadcast(
 		return "", errors.Wrap(err, "unable to encode tx")
 	}
 
+	// skip broadcasting if mempool is already congested
+	resp, err := c.cometBFTClient.NumUnconfirmedTxs(ctx)
```

Comment:
Can't we rather have a separate process that fetch the value into the context, so we don't query for every broadcast